discurses/ui/message_list.py

```diff
@@ -401,6 +405,35 @@ def update_columns(*args, **kwargs):
         pass
 
 
+class ForbiddenWidget(urwid.WidgetWrap):
+    """This widget will be displayed in channels you can't access"""
+
+    def __init__(self, chat_widget):
+        self.chat_widget = chat_widget
+        self.message = FakeMessage(datetime.datetime.min)
+        self._selectable = False
+        txt = urwid.Text(
+            "                                                               \n"
+            "                                                               \n"
+            "                                                               \n"
+            "                                                               \n"
+            "< moo >                                                        \n"
+            " -----                                                         \n"
+            "        \   ^__^                                               \n"
+            "         \  (oo)\_______                                       \n"
+            "            (__)\       )\/\                                   \n"
+            "                ||----w |                                      \n"
+            "                ||     ||                                      \n"
+            "                                                               \n"
+            "                                                               \n"
+            "Oops! This channel is Forbidden. You don't have access. Sorry!\n\n",
+            align=urwid.CENTER)
+        w = urwid.Padding(txt, left=5, right=5)
+        self.__super.__init__(w)
+
+    def update_columns(*args, **kwargs):
```

Comment:
What is this method for?

Reply:
See line 87. This widget acts as a MessageWidget, and those need an update_columns function.

Reply:
I see. If there is nothing to do in this `update_columns` method, I think that you can discard it in the child class.

Reply:
There is no common `MessageLikeWidget` superclass, it uses duck typing

Reply:
Oh I see! Let's keep this then, and consider refactoring the widget's hierarchy in the future.

Reply:
Fixed the PEP-8 issues, now passes CI.

Reply:
Thanks a lot! I'll try to test to feature asap & merge! ;)

Reply:
Mmh, I don't know how you can connect to a forbidden channel through discurses. Is there any way to manually type a chan name that is hidden (& forbidden) to reach your code?

Reply:
if its the default channel - see #29

Reply:
When I joined a server with a Forbidden channel, it showed up in the channel list in the server tree.

Reply:
It wasn't the default channel on the server, in this case.

Reply:
Ok

@topisani I can't test @glitchassassin atm. Shall I merge still?